test/circuit_breaker_test.rb

```diff
@@ -358,18 +358,50 @@ def test_error_threshold_timeout_is_skipped_when_not_using_error_threshold_and_n
 
   def test_env_var_disables_circuit_breaker
     ENV["SEMIAN_CIRCUIT_BREAKER_DISABLED"] = "1"
-    open_circuit!
+    resource = Semian.register(
+      :env_var_disables_circuit_breaker,
+      tickets: 1,
+      error_threshold: 1,
+      error_timeout: 10,
+      success_threshold: 1,
+    )
+    open_circuit!(resource)
 
-    assert_circuit_closed
+    assert_circuit_closed(resource)
   ensure
     ENV.delete("SEMIAN_CIRCUIT_BREAKER_DISABLED")
   end
 
-  def test_semian_wide_env_var_disables_circuit_breaker
+  def test_semian_wide_env_var_does_not_create_resource
     ENV["SEMIAN_DISABLED"] = "1"
-    open_circuit!
 
-    assert_circuit_closed
+    resource = Semian.register(
+      :global_env_var_disables_circuit_breaker,
+      tickets: 1,
+      error_threshold: 2,
+      error_timeout: 5,
+      success_threshold: 1,
+    )
+
+    assert_nil(resource)
+  ensure
+    ENV.delete("SEMIAN_DISABLED")
+  end
+
+  def test_semian_wide_env_disables_totaly_even_with_explicit_circuit_breaker
+    ENV["SEMIAN_DISABLED"] = "1"
+
+    resource = Semian.register(
+      :global_env_var_disables_circuit_breaker,
+      bulkhead: true,
+      tickets: 1,
+      circuit_breaker: true,
+      error_threshold: 2,
+      error_timeout: 5,
+      success_threshold: 1,
+    )
+
+    assert_nil(resource)
```

Comment:
Looks like we always returned a non-nil value from `.register` before (or raised an exception). Is this method exposed to apps using Semian?

If so, and I'm reading it correctly, this may not be backwards-compatible.

Reply:
Mostly. The problem that before it would not be raise, but NOOP. Because it created always resource objects, and skip logic during on acquire.

Reply:
That is true. Need may be stress it in changelog and also in gemspec post install notifications.